### DISCUSSION TITLE
직렬화, 역직렬화 (serialize와 deserialize)

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#576] [5. Q&A]
https://docs.unity3d.com/kr/2020.3/Manual/JSONSerialization.html
serialize와 deserialize
다음과 같이 직렬화, 역직렬화를 테스트 해보고 있습니다 

MyPacket클래스를 만들고 
![image](https://user-images.githubusercontent.com/50188795/161552990-88f723e2-ed7f-48af-b3ad-0ecb0fd434be.png)

import한후 
![image](https://user-images.githubusercontent.com/50188795/161553061-f1bc6713-ebe8-4e1f-a1b0-d86b6390679c.png)

인스턴스를 생성후 필드에 값을 넣어 주고 
![image](https://user-images.githubusercontent.com/50188795/161553286-4b6b085f-6d97-400e-b66c-480c1e02dfae.png)

직렬화를 시도 해보았습니다 
![image](https://user-images.githubusercontent.com/50188795/161553360-499b48dd-ecc2-4e69-9018-4ffe6eb08ce4.png)

다음과 같은 에러가 나옵니다 
![image](https://user-images.githubusercontent.com/50188795/161553399-27f562f3-63eb-4052-8d55-74122ce9ff5d.png)

다음과 같이 시도해 보았습니다 
![image](https://user-images.githubusercontent.com/50188795/161554720-bf0cd25e-ec3b-4b80-85b4-4bb67b3996e4.png)

같은 에러가 발생 했습니다 
![image](https://user-images.githubusercontent.com/50188795/161554691-857dcb25-0286-47f4-8f2c-c655d6f8a63b.png)

매개변수가 안맞다는데 
![image](https://user-images.githubusercontent.com/50188795/161554760-bc3b4fce-0e64-443f-9919-92c0f8f50ffd.png)

![image](https://user-images.githubusercontent.com/50188795/161554843-37255260-9c5f-4d25-b8b1-f86a4e1a5bb2.png)

역직렬화도 시도해보았습니다 
![image](https://user-images.githubusercontent.com/50188795/161555639-c837f45a-0442-4f35-9da4-cf735b395a7c.png)

다음과 같은 에러가 납니다 
![image](https://user-images.githubusercontent.com/50188795/161555768-e2a97821-fa86-4e15-963b-bc4b3572a17a.png)

무엇을 잘못하고 있는것일까요?

newtonsoft json은 사용못하는것인가요?

https://www.newtonsoft.com/json

------------------------------------------------------------

그냥 js에서 하던방식대로 하면 되는군요 >_<

![image](https://user-images.githubusercontent.com/50188795/161556951-8d41c0e4-decc-43a4-870f-2e642a29db15.png)

![image](https://user-images.githubusercontent.com/50188795/161557042-9859322d-f205-44ea-8055-9a2af1a9e163.png)

**Comment by x4cf** [comment]
좋은 꿀팁이군요! 메모메모!
